tests/lms/extensions/feature_flags/__init___test.py

```diff
@@ -0,0 +1,53 @@
+from unittest import mock
+
+import pytest
+from pyramid.interfaces import IRequestExtensions
+
+from lms.extensions.feature_flags import includeme
+
+
+class TestIncludeMe:
+    def test_it_creates_the_feature_flags_object(self, pyramid_config, FeatureFlags):
+        includeme(pyramid_config)
+
+        FeatureFlags.assert_called_once_with()
+
+    def test_feature(self, feature_flags, pyramid_config, pyramid_request):
+        includeme(pyramid_config)
+        feature = pyramid_config.registry.queryUtility(IRequestExtensions).methods[
+            "feature"
+        ]
+
+        feature(pyramid_request, "test_feature")
+
+        feature_flags.flag_is_active.assert_called_once_with(
+            pyramid_request, "test_feature"
+        )
+
+    def test_add_feature_flag_provider(self, pyramid_config, feature_flags):
```

Comment:
I wonder if we really need both the singular and plural APIs. If you want to steer users of the package towards registering all the providers together so that the order of precedence is obvious, it might make sense to keep just the plural API.

Reply:
I think it just makes it very slightly easier to use. If you have multiple to add one call is easier. If you have just one to add, then `add_feature_flag_provider(func)` is more intuitive than `add_feature_flag_providers(func)`